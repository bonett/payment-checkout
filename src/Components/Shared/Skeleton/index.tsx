/* eslint-disable */
import React from "react";

function Skeleton() {
    return (
        <div className="skeleton-content">
            {
                [1, 2, 3, 4, 5, 6].map(i => (
                    <article key={i} className="article-content skeleton-wrapper">
                        <div className="article-content_thumb">
                            <div className="skeleton skeleton-img"></div>
                        </div>
                        <div className="article-content_description">
                            <div className="skeleton skeleton-text"></div>
                            <div className="skeleton skeleton-text"></div>
                            <div className="skeleton skeleton-text"></div>
                            <div className="skeleton skeleton-text"></div>
                        </div>
                    </article>
                ))
            }
        </div>
    );
}

export default Skeleton;
