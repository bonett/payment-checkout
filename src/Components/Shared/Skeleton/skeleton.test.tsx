import React from "react";
import { render, screen } from "@testing-library/react";
import Skeleton from "./";

describe("Skeleton component", () => {
  test("Should renders skeleton component", () => {
    const { container } = render(<Skeleton />);
  
    const cards = screen.getAllByTestId("skeleton-card");
    expect(cards.length).toEqual(6);
    expect(container).toMatchSnapshot();
  });
});
