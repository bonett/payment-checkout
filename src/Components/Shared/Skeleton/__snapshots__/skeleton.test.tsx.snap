// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton component Should renders skeleton component 1`] = `
<div>
  <div
    class="skeleton-content"
  >
    <article
      class="article-content skeleton-wrapper"
      data-testid="skeleton-card"
    >
      <div
        class="article-content_thumb"
      >
        <div
          class="skeleton skeleton-img"
        />
      </div>
      <div
        class="article-content_description"
      >
        <div
          class="skeleton skeleton-text"
        />
        <div
          class="skeleton skeleton-text"
        />
        <div
          class="skeleton skeleton-text"
        />
        <div
          class="skeleton skeleton-text"
        />
      </div>
    </article>
    <article
      class="article-content skeleton-wrapper"
      data-testid="skeleton-card"
    >
      <div
        class="article-content_thumb"
      >
        <div
          class="skeleton skeleton-img"
        />
      </div>
      <div
        class="article-content_description"
      >
        <div
          class="skeleton skeleton-text"
        />
        <div
          class="skeleton skeleton-text"
        />
        <div
          class="skeleton skeleton-text"
        />
        <div
          class="skeleton skeleton-text"
        />
      </div>
    </article>
    <article
      class="article-content skeleton-wrapper"
      data-testid="skeleton-card"
    >
      <div
        class="article-content_thumb"
      >
        <div
          class="skeleton skeleton-img"
        />
      </div>
      <div
        class="article-content_description"
      >
        <div
          class="skeleton skeleton-text"
        />
        <div
          class="skeleton skeleton-text"
        />
        <div
          class="skeleton skeleton-text"
        />
        <div
          class="skeleton skeleton-text"
        />
      </div>
    </article>
    <article
      class="article-content skeleton-wrapper"
      data-testid="skeleton-card"
    >
      <div
        class="article-content_thumb"
      >
        <div
          class="skeleton skeleton-img"
        />
      </div>
      <div
        class="article-content_description"
      >
        <div
          class="skeleton skeleton-text"
        />
        <div
          class="skeleton skeleton-text"
        />
        <div
          class="skeleton skeleton-text"
        />
        <div
          class="skeleton skeleton-text"
        />
      </div>
    </article>
    <article
      class="article-content skeleton-wrapper"
      data-testid="skeleton-card"
    >
      <div
        class="article-content_thumb"
      >
        <div
          class="skeleton skeleton-img"
        />
      </div>
      <div
        class="article-content_description"
      >
        <div
          class="skeleton skeleton-text"
        />
        <div
          class="skeleton skeleton-text"
        />
        <div
          class="skeleton skeleton-text"
        />
        <div
          class="skeleton skeleton-text"
        />
      </div>
    </article>
    <article
      class="article-content skeleton-wrapper"
      data-testid="skeleton-card"
    >
      <div
        class="article-content_thumb"
      >
        <div
          class="skeleton skeleton-img"
        />
      </div>
      <div
        class="article-content_description"
      >
        <div
          class="skeleton skeleton-text"
        />
        <div
          class="skeleton skeleton-text"
        />
        <div
          class="skeleton skeleton-text"
        />
        <div
          class="skeleton skeleton-text"
        />
      </div>
    </article>
  </div>
</div>
`;
